pkg/storagev2/storagetest/storage.go

```diff
@@ -347,6 +347,28 @@ func TestStore(t *testing.T, s storage.Store) {
 		})
 	})
 
+	t.Run("iterate start prefix", func(t *testing.T) {
+		t.Run("obj1", func(t *testing.T) {
```

Comment:
It would be good if we can have some other prefix entries also in the database and ensure that they are not being iterated in the iteration.